crates/rust-analyzer/src/lsp_ext.rs

```diff
@@ -177,6 +177,26 @@ pub struct CargoRunnable {
     pub expect_test: Option<bool>,
 }
 
+pub enum RelatedTests {}
+
+impl Request for RelatedTests {
+    type Params = RelatedTestsParams;
```

Comment:
```suggestion
    type Params = lsp_types::TextDocumentPositionParams;
```

Reply:
I plan to expand the RelatedTestsParams in the future.
For example to support https://github.com/microsoft/vscode/issues/107467. I saw the https://github.com/rust-analyzer/rust-analyzer/pull/5765 but seems it's abandoned. And in my opinion VSCode native tests support is preferable.

Reply:
Perhaps `RelatedTests` is not the best name for the request, but just `Tests` is even worse, in my opinion. Any ideas?

Reply:
>I plan to expand the RelatedTestsParams in the future.

We'll add `RelatedTestsParams` with serde flatten at that point 

>Perhaps RelatedTests is not the best name for the request,

it sounds ok to me tbh. We can always change later. 


And yeah, it'd be lovely to switch to upstream impl for running, debugging and testing, but we realistically can do that only when upstream supports "run thing at point" workflow. Not sure what's the status there.

Reply:
> I plan to expand the RelatedTestsParams in the future.
> For example to support [microsoft/vscode#107467](https://github.com/microsoft/vscode/issues/107467). I saw the #5765 but seems it's abandoned. And in my opinion VSCode native tests support is preferable.

It has been requested, but for now, I am finishing my diploma and then I will return to the project. These are not tests. It's Runnable. That is, tests, examples, entry functions. If you want, you can implement an approximate date when I plan to return to it at the beginning of the next month.

Reply:
> I plan to expand the RelatedTestsParams in the future.
> For example to support [microsoft/vscode#107467](https://github.com/microsoft/vscode/issues/107467). I saw the #5765 but seems it's abandoned. And in my opinion VSCode native tests support is preferable.

Hey @vsrs - thanks for this awesome feature! 

I am implementing the emacs lisp lsp-mode client code for this extension (my WIP PR here https://github.com/emacs-lsp/lsp-mode/pull/2776) and i wanted to clarify why you decided to return `TestInfo[]` as a wrapper over `Runnable` instead of returning `Runnable[]`? Using the same interface as already implemented runnable methods makes it easier to integrate now and enables future code re-use. 

Do you intent to expand change the protocol and add more keys to TestInfo? If not, would you and @matklad be ok if I open a PR that changed the Request surface for RelatedTests (and all the corresponding front-end callsites?)

```rust
impl Request for RelatedTests {
    type Params = lsp_types::TextDocumentPositionParams;
    type Result = Vec<Runnable>;
}
```

Reply:
> Do you intent to expand change the protocol and add more keys to TestInfo?
Yes, I do. I'm working on a Rust test provider\adapter using upcoming VSCode API ( https://github.com/microsoft/vscode/issues/107467) and I reuse expanded TestInfo.

Reply:
I don't use vscode but i would like to bring as much OOTB goodness from vscode as possible, so would appreciate your input. 

The link you provided is very comprehensive and since I cannot ctrl-f `TestInfo` anywhere in the document, can you please provide some detail. where and how is testinfo going to be used?

just for me to understand - will you be modifying this rust-analyzer endpoint or do you intend to add new ones? If the former, will you be breaking backwards compat of the interface and what fields do you want to add?

Reply:
The current VSCode test adapter API design is host only and hasn't LSP part, so `TestInfo` is my own abstraction and it's only a draft. I cannot say right now how it will be changed in the future. But as every test is a runnable, `TestInfo` will always have the `pub runnable: Runnable,` field. I expect that `rust-analyzer/relatedTests` request will not use new fields\structures, that is why I decided to publish this feature before the test adapter itself.